source/fab/tasks/fortran.py

```diff
@@ -303,9 +310,9 @@ def line_by_line(self) -> Iterator[str]:
             self._line_buffer = ''
 
 
-class FortranAnalyser(Analyser):
-    def __init__(self, reader: TextReader, database: SqliteStateDatabase):
-        super().__init__(reader, database)
+class FortranAnalyser(Task):
+    def __init__(self, workspace: Path):
+        self.database = SqliteStateDatabase(workspace)
```

Comment:
Workspace is only used for instantiating the database. If the level above already has the database why not pass that instead?

Reply:
I was thinking that eventually we won't want to have the block in `builder.py` which prints out the database contents to `stdout` - and since that is all that the top level uses the database for at this point we might be able to remove that from the top level.  Happy to change it though, what do you think?

Reply:
That sounds reasonable. Lets just keep an eye on it so we're not backing life complicated for ourselves. No further changes required for the moment.